index.bs

```diff
@@ -5953,6 +5935,54 @@ and joining them with the string “Or”.
 <div data-fill-with="grammar-NonAnyType"></div>
 
 
+<h4 id="idl-types-with-extended-attributes">Types with extended attributes</h4>
+
+Additional types can be created from existing ones by specifying certain [=extended attributes=] on
+the existing types. For example, <code>[Clamp] long</code> defines a new type whose behavior is
+based on that of the {{long}} type, but modified as specified by the [{{Clamp}}] extended attribute.
+
+<p id="extended-attributes-applicable-to-types">
+    The following extended attributes are applicable to types:
+    [{{Clamp}}],
+    [{{EnforceRange}}], and
+    [{{LegacyTreatNullAsEmptyString}}].
+</p>
+
+<div algorithm>
+    The <dfn for="IDL type" lt="extended attribute associated with|extended attributes associated with">extended attributes associated with</dfn>
+    a [=type=] |type| are determined as follows:
+
+    1.  Let |extended attributes| be the empty set.
+    1.  If |type| appears as part of a
+        <emu-nt><a href="#prod-TypeWithExtendedAttributes">TypeWithExtendedAttributes</a></emu-nt>
+        production, add the [=extended attributes=] present in the production's
+        <emu-nt><a href="#prod-ExtendedAttributeList">ExtendedAttributeList</a></emu-nt> to
+        |extended attributes|.
+    1.  If |type| is a [=member type=] of a [=union type=] |U|, add the
+        [=extended attributes associated with=] |U| to |extended attributes|.
+    1.  If |type| appears as part of a <emu-nt><a href="#prod-Type">Type</a></emu-nt> production
+        within an <emu-nt><a href="#prod-Argument">Argument</a></emu-nt> production, add to
+        |extended attributes| all of the [=extended attributes=] present in the production's
+        <emu-nt><a href="#prod-ExtendedAttributeList">ExtendedAttributeList</a></emu-nt> that are
+        <a href="#extended-attributes-applicable-to-types">applicable to types</a>.
+    1.  If |type| appears as part of a <emu-nt><a href="#prod-Type">Type</a></emu-nt> production
+        within an <emu-nt><a href="#prod-DictionaryMember">DictionaryMember</a></emu-nt> production,
+        add to |extended attributes| all of the [=extended attributes=] present in the production's
+        <emu-nt><a href="#prod-ExtendedAttributeList">ExtendedAttributeList</a></emu-nt> that are
+        <a href="#extended-attributes-applicable-to-types">applicable to types</a>.
+    1.  If |type| is a [=typedef=], add the [=extended attributes associated with=] the type being
+        given a new name to |extended attributes|.
+    1. Return |extended attributes|.
+</div>
+
+For any [=type=], the [=extended attributes associated with=] it must only contain
+[=extended attributes=] that are
+<a href="#extended-attributes-applicable-to-types">applicable to types</a>.
+
+The [=type name=] of a type associated with [=extended attributes=] is the concatenation of the
```

Comment:
OK, I can't add an example since it's actually impossible to use any of the current type-applicable extended attributes with nullable types, or use them together.